src/hooks/screen/useCreateChat.ts

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import {getOrCreateAnonymousChannel} from '../../service/chat';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import ChannelList from '../../database/schema/ChannelListSchema';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import UserSchema from '../../database/schema/UserSchema';
+import ChannelListMemberSchema from '../../database/schema/ChannelListMemberSchema';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import {GROUP_INFO} from '../core/constant';
+
+const useCreateChat = () => {
+  const [loadingCreateChat, setLoadingCreateChat] = React.useState(false);
+  const {localDb} = useLocalDatabaseHook();
+  const {goToChatScreen} = useChatUtilsHook();
+  const createSignChat = async (members: string[], selectedUser, from) => {
+    try {
+      setLoadingCreateChat(true);
+      const initChannel = await SignedMessageRepo.createSignedChat(members);
+      const chatData = await createChannelJson(initChannel, selectedUser);
+
+      const channelList = await ChannelList.fromMessageSignedAPI(chatData);
+      channelList.saveIfLatest(localDb);
+      handleMemberSchema(initChannel);
+      setLoadingCreateChat(false);
+      goToChatScreen(channelList, from);
+    } catch (e) {
+      setLoadingCreateChat(false);
+      console.log({e}, 'error create chat');
+    }
+  };
+
+  const createChannelJson = (response, selectedUser) => {
+    const channelWithMember = {...response.channel, members: response.members};
+    const targetRawJson = {
+      type: 'notification.message_new',
+      cid: response?.channel?.id,
+      channel_id: '',
+      channel_type: 'messaging',
+      channel: channelWithMember,
+      created_at: response?.channel,
+      targetName: selectedUser?.user?.name,
+      targetImage: selectedUser?.user?.image
+    };
+    const chatData = {
+      channel: response?.channel,
+      members: response?.members,
+      appAdditionalData: {
+        rawJson: targetRawJson,
+        message: '',
+        targetName: selectedUser?.user?.name,
+        targetImage: selectedUser?.user?.image
+      }
+    };
+    return chatData;
+  };
+
+  const handleMemberSchema = (response) => {
+    try {
+      response?.members?.forEach(async (member) => {
+        const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
+        await userMember.saveOrUpdateIfExists(localDb);
+
+        const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+          response?.channel?.id,
+          uuid(),
+          member
+        );
+        await memberSchema.save(localDb);
+      });
+    } catch (e) {
+      console.log(e, 'error on memberSchema');
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleMemberSchema` function, you are using `forEach` with an async function. This can lead to unexpected behavior because `forEach` doesn't wait for promises. Instead, consider using `for...of` loop or `Promise.all()` with `map()` to handle asynchronous operations within loops.

```diff
    try {
-       response?.members?.forEach(async (member) => {
+       const memberPromises = response?.members?.map(async (member) => {
        const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
        await userMember.saveOrUpdateIfExists(localDb);

        const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
          response?.channel?.id,
          uuid(),
          member
        );
        await memberSchema.save(localDb);
-       });
+       });
+       await Promise.all(memberPromises);
    } catch (e) {
      console.log(e, 'error on memberSchema');
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->